private/pkg/transport/http2client/proxy.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http2client
+
+import (
+	"bufio"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+)
+
+// This file was adapted from https://github.com/grpc/grpc-go/blob/master/internal/transport/proxy.go
```

Comment:
This looks good to me in theory, but we should sanity check with responsible people on what conditions we need to meet - we're already using APACHE license on buf, but do we need to eg use the same license header from grpc-go?

Reply:
Here's an example of how we reference other licenses when appropriate https://github.com/bufbuild/buf/blob/main/private/pkg/diff/diff.go#L22=